docs/faucets/dd.md

```diff
@@ -0,0 +1,366 @@
+
+The benefit is that if we went with the signup route, we could advertise to those users.
+
+The downside is that that is significantly more infrastructure, and it doesn't necessarily improve the safety/control of the system.
+
+### Staking Asset
+
+The staking asset will be deployed with the same owner as the fee asset. In addition, a `StakingAssetHandler` will be deployed, which will have the minter role on the staking asset.
+
+### Staking Asset Handler
+
+```solidity
+interface IStakingAssetHandler {
+	function addValidator(address _attester, address _proposer) external;
+	function setRollup(address _rollup) external;
+	function setDepositAmount(uint256 _amount) external;
+	function setMinMintInterval(uint256 _interval) external;
+}
+
+contract StakingAssetHandler is IStakingAssetHandler, Ownable {
+	IMintableERC20 public immutable STAKING_ASSET;
+
+	uint256 public depositAmount;
+	uint256 public lastMintTimestamp;
+	uint256 public minMintInterval;
+	IStakingCore public rollup;
+
+	constructor(address _owner, address _stakingAsset, address _rollup, uint256 _depositAmount, uint256 _minMintInterval) Ownable(_owner) {
+		STAKING_ASSET = IMintableERC20(_stakingAsset);
+		depositAmount = _depositAmount;
+		rollup = IStakingCore(_rollup);
+		minMintInterval = _minMintInterval;
+	}
+
+	function addValidator(address _attester, address _proposer) external override onlyOwner {
+		require(block.timestamp - lastMintTimestamp >= minMintInterval, "Min mint interval not met");
+		lastMintTimestamp = block.timestamp;
+		STAKING_ASSET.mint(address(this), depositAmount);
+		rollup.deposit(_attester, _proposer, owner(), depositAmount);
+	}
+
+	function setRollup(address _rollup) external override onlyOwner {
+		rollup = IStakingCore(_rollup);
+	}
+
+	function setDepositAmount(uint256 _amount) external override onlyOwner {
+		depositAmount = _amount;
+	}
+
+	function setMinMintInterval(uint256 _interval) external override onlyOwner {
+		minMintInterval = _interval;
+	}
+
+}
+```
+
+The owner of the handler will be a different EOA, since it will be held by the discord bot.
+
+#### Upgrades
+
+The handler will need to be upgraded to the new rollup address when a new rollup is made canonical.
+
+This will be done by manually by the person on the Aztec Labs team managing the discord bot.
+
+### Discord Bot
+
+The `/add-validator` command needs to be augmented to accept a block number and an archive merkle proof.
+
+When this command is invoked, it should:
+
+1. Ensure the user has not added `X` validators already (using a mapping based on the user's discord ID)
+2. Query L1 for the proven tip of the chain
+3. Ensure that proven tip's block number matches the block number provided by the user
+4. Verify the merkle proof
+5. Mint staking assets for itself
+6. Add the user's address to the validator set, specifying the discord bot's address as the withdrawer
```

Comment:
As above, I don't think we should have the bot as owner and withdrawer and all. Don't want anything remotely close to admin rights in the hottest of all wallets if we don't need to.